ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/location/CaptureLocationTaskFragment.kt

```diff
@@ -50,6 +52,15 @@ class CaptureLocationTaskFragment : AbstractTaskFragment<CaptureLocationTaskView
     return rowLayout
   }
 
+  override fun onTaskResume() {
+    // Ensure that the location lock is enabled, if it hasn't been.
+    if (viewModel.enableLocationLockFlow.value == LocationLockState.NEEDS_ENABLE) {
+      viewLifecycleOwner.lifecycleScope.launch {
+        viewModel.enableLocationLockFlow.emit(LocationLockState.ENABLE)
+      }
+    }
```

Comment:
We can simplify this by creating a `enableLocationLock()` method in viewModel and moving all these details into that. Then we won't have to expose any flows or other states to this fragment. Thoughts?

Reply:
This is a great suggestion, I was even able to refactor the login in `onMapReady()` and set `enableLocationLockFlow` to private, which is nice separation of concerns. Done.